### ISSUE TITLE
Databricks spark 3.0 SQL dbt-date support 

### ISSUE THREAD

**Issue by pa-uriza** [calogica/dbt-date#119]
### Is this a new bug in dbt-date?

- [x] I believe this is a new bug in dbt-date
- [x] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior
When running dbt-date in a Databricks environment I get the following error:

```
[INCONSISTENT_BEHAVIOR_CROSS_VERSION.DATETIME_PATTERN_RECOGNITION] You may get a different result due to the upgrading to Spark >= 3.0:
Fail to recognize 'LLLL' pattern in the DateTimeFormatter. 1) You can set "legacy_time_parser_policy" to "LEGACY" to restore the behavior before Spark 3.0. 2) You can form a valid datetime pattern with the guide from 'https://spark.apache.org/docs/latest/sql-ref-datetime-pattern.html'.
```

This happens when the macro [`spark__month_name`](https://github.com/calogica/dbt-date/blob/df2f567efacd1e0074f86600812213f5d79c1b1b/macros/calendar_date/month_name.sql#L35C1-L38C16) is called.

I tested the following SQL in Databricks environment and I get the same error

```sql
select date_format(now(), 'LLL');
```

```sql
select date_format(now(), 'LLLL');
```

### Expected Behavior

Get the month name without errors. This could be obtained for new spark versions with either

```sql
select date_format(now(), 'MMM');
```

or

```sql
select date_format(now(), 'MMMM');
```

### Steps To Reproduce

Run dbt-date

### Relevant log output

```
[INCONSISTENT_BEHAVIOR_CROSS_VERSION.DATETIME_PATTERN_RECOGNITION] You may get a different result due to the upgrading to Spark >= 3.0:
Fail to recognize 'LLLL' pattern in the DateTimeFormatter. 1) You can set "legacy_time_parser_policy" to "LEGACY" to restore the behavior before Spark 3.0. 2) You can form a valid datetime pattern with the guide from 'https://spark.apache.org/docs/latest/sql-ref-datetime-pattern.html'.
```

### Environment

```markdown
- OS: Linux
- Python: 3.10
- dbt: 1.7.4
```

### Which database adapter are you using with dbt?

Databricks

**Comment by flixflop** [comment]
I have encountered the same issue. What I do not quite understand is that the Spark documentation claims that either 'M' or 'L' should work.
But I found that when adjusting the the spark__month_name macro to use 'M' instead of 'L' then the issue seems to be resolved.

**Comment by pa-uriza** [comment]
Solved with the PR #123
